src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java

```diff
@@ -134,6 +135,34 @@ public static User parse(XContentParser parser) throws IOException {
         return new User(name, backendRoles, roles, customAttNames);
     }
 
+    /**
+     * User String format must be pipe separated as : user_name|backendrole1,backendrole2|roles1,role2
```

Comment:
We should not allow user input "|" in user/role name. Do we have such checking ?

Reply:
Security plugin has those checks, we use similar format for UserInjection and RolesInjection
https://github.com/opendistro-for-elasticsearch/security/blob/6304e7875ba7dc24ffcb0fe89a875a56cb6cc853/src/main/java/com/amazon/opendistroforelasticsearch/security/auth/RolesInjector.java#L35
https://github.com/opendistro-for-elasticsearch/security/blob/b69e821a360a774ec2e94dc01390cd61521562d2/src/main/java/com/amazon/opendistroforelasticsearch/security/auth/UserInjector.java#L84

Reply:
Related PR: https://github.com/opendistro-for-elasticsearch/security/pull/798